subtrack.MAUI/Services/SubscriptionsCalculator.cs

```diff
@@ -56,6 +56,26 @@ public IEnumerable<Subscription> GetSubscriptionListByMonth(IEnumerable<Subscrip
 
     public DateTime GetNextPaymentDate(Subscription subscription)
     {
-        return subscription.LastPayment.AddMonths(1).Date;
+
+        var nextPaymentDate = subscription.LastPayment.AddMonths(1).Date;
```

Comment:
I *think* this logic can be simplified. Assuming that leap year is already implemented in the DateTime.DaysInMonth and .AddMonth 

```
// next payment date without considering the starting day
    var nextMonthDate = lastPayment.AddMonths(1);
    var nextMonthTotalDays = DateTime.DaysInMonth(nextMonthDate.Year, nextMonthDate.Month);

// if we started on a month that has more days than the next month we simply want to get the last day of the next month 
    if (startDay > nextMonthTotalDays)
    {
        //return new DateTime(nextMonthDate.Year, nextMonthDate.Month, nextMonthTotalDays);
// next monthdate gets the last day of the month if the day requested does not exist so the line above is not needed
          return nextMonthDate;
    }

// if the next month has more days or the same amount of days as the starting day we want to get the starting day
    return new DateTime(nextMonthDate.Year, nextMonthDate.Month, startDay);
```

Reply:
coudn't make the code pass all test cases without the additional conditional statement on line 72 from latest commit 
maybe im not undertanding fully...

Reply:
I checked what tests were failing with the logic i posted above:
```
yield return new object[] { new DateTime(2023, 9, 30), new DateTime(2023, 10, 31) };
yield return new object[] { new DateTime(2023, 10, 31), new DateTime(2023, 11, 30) };
```

Theese tests are incorrect, 30/9 should be 30/10. 31/10 should be 30/11 _unless_ FirstPaymentDate is set to 31.